src/compiler/crystal/semantic/call_error.cr

```diff
@@ -282,6 +285,19 @@ class Crystal::Call
     raise message, owner_trace
   end
 
+  def logical_operator?(def_name)
```

Comment:
created these methods to avoid getting the `raise_matches_not_found` method getting bigger and bigger.

If makes sense to keep them, my idea is to keep extracting more methods (is is a good idea?)